sherpa-onnx/csrc/offline-transducer-greedy-search-decoder.cc

```diff
@@ -46,7 +46,10 @@ OfflineTransducerGreedySearchDecoder::Decode(Ort::Value encoder_out,
     start += n;
     Ort::Value logit = model_->RunJoiner(std::move(cur_encoder_out),
                                          std::move(cur_decoder_out));
-    const float *p_logit = logit.GetTensorData<float>();
+    float *p_logit = logit.GetTensorMutableData<float>();
+    if (blank_penalty_ > 0.0) {
+      p_logit[0] -= blank_penalty_; // assuming blank id is 0
```

Comment:
Shall we also consider the case when `batch_size > 1` ?

`p_logit[0]` is only for the first utterance.

 We need to process
```bash
p_logit[vocab_size*i + 0]
// for in range(n)
```

You can move this `if` statement into the `for` statement below.

Reply:
Fixed!